### DISCUSSION TITLE
Error while hosting on render, didn't able to connect with io server( or didn't know how to connect)

### DISCUSSION THREAD

**Discussion by Vansh1190** [socketio/socket.io#4771] [Q&A] [answered]
I am building a chat application with socket.io and for the routes i used express router and
For realtime chatting i used socket.io.
The application is running very good in my local server,
My backend is running on port 3000
And my socket is running on port 3002
But when i hosted it on render, it gives me an error that,
The frontend is not able to connect with my socket.
I used io.connect(https://renderUrl.com:3002)
Not working!
Here's the link to my repo. - https://github.com/Vansh1190/chatverse-backend

Please help its very urgent

**Comment by darrachequesne** [comment, marked as answer]
Hi! It seems that you are not able to reach the Socket.IO server. Could you please check if the port 3002 is open on your hosting platform? Or could you switch to 443?

See also: https://socket.io/docs/v4/troubleshooting-connection-issues/

**Reply by Vansh1190** [reply]
Sir the Problem is now resolved, Thank you for reply

**Reply by oluwaseun-demonflowne** [reply]
how were you able to host a socket on render

**Reply by BhanuPratap0** [reply]
how you resolved it?

**Reply by chidubemOkafor** [reply]
when you have a solution to your problem you should tell.

**Reply by tomomoto** [reply]
Sir uses a node.js as a server hosting http server and io server. ExpressJS and Socket.IO can reuse one port.
Did the same thing many years ago and it was working.
https://doc.scalingo.com/languages/nodejs/websocket-web-same-port
